src/ext/java/org/opentripplanner/ext/legacygraphqlapi/datafetchers/LegacyGraphQLQueryTypeImpl.java

```diff
@@ -909,15 +909,19 @@ public DataFetcher<Iterable<Route>> routes() {
       if (args.getLegacyGraphQLName() != null) {
         String name = args.getLegacyGraphQLName().toLowerCase(environment.getLocale());
         routeStream =
-          routeStream.filter(route ->
-            Stream
-              .of(route.getShortName(), route.getLongName())
-              .filter(Objects::nonNull)
-              .map(s -> s.toLowerCase(environment.getLocale()))
-              .anyMatch(s -> s.startsWith(name))
-          );
+          routeStream.filter(route -> {
+            if (
+              route.getShortName() != null &&
+              route.getShortName().toLowerCase(environment.getLocale()).startsWith(name)
+            ) {
+              return true;
+            }
+            return (
+              route.getLongName() != null &&
+              route.getLongName().toString(environment.getLocale()).toLowerCase().startsWith(name)
```

Comment:
Is it intentional that the call to `toLowerCase` is slightly different for longName and shortName?

If no, then this could be made into a reusable function.

Reply:
Shortname does not have translations, so it does differ little bit from longName call. But I created a reusable method as well.